google/cloud/aiplatform/training_jobs.py

```diff
@@ -396,33 +429,89 @@ def _create_input_data_config(
                 -  AIP_TRAINING_DATA_URI ="bigquery_destination.dataset_*.training"
                 -  AIP_VALIDATION_DATA_URI = "bigquery_destination.dataset_*.validation"
                 -  AIP_TEST_DATA_URI = "bigquery_destination.dataset_*.test"
+        Raises:
+            ValueError: When more than 1 type of split configuration is passed or when
+                the split configuartion passed is incompatible with the dataset schema.
         """
 
         input_data_config = None
         if dataset:
-            # Create fraction split spec
-            fraction_split = gca_training_pipeline.FractionSplit(
-                training_fraction=training_fraction_split,
-                validation_fraction=validation_fraction_split,
-                test_fraction=test_fraction_split,
-            )
+            # Initialize all possible splits
+            filter_split = None
+            predefined_split = None
+            timestamp_split = None
+            fraction_split = None
+
+            # Create filter split
+            filter_split = None
+            if all(
+                [
+                    training_filter_split is not None,
+                    validation_filter_split is not None,
+                    test_filter_split is not None,
+                ]
+            ):
+                filter_split = gca_training_pipeline.FilterSplit(
+                    training_filter=training_filter_split,
+                    validation_filter=validation_filter_split,
+                    test_filter=test_filter_split,
+                )
 
-            # Create predefined split spec
+            # Create predefined split
             predefined_split = None
             if predefined_split_column_name:
-                if dataset._gca_resource.metadata_schema_uri not in (
-                    schema.dataset.metadata.tabular,
-                    schema.dataset.metadata.time_series,
-                ):
-                    raise ValueError(
-                        "A pre-defined split may only be used with a tabular or time series Dataset"
-                    )
-
                 predefined_split = gca_training_pipeline.PredefinedSplit(
                     key=predefined_split_column_name
                 )
 
-            # Create GCS destination
+            # Create timestamp split or fraction split
+            if all(
+                [
+                    training_fraction_split is not None,
+                    validation_fraction_split is not None,
+                    test_fraction_split is not None,
+                ]
+            ):
+                if timestamp_split_column_name:
```

Comment:
This nested block wouldn't catch the case where `timestamp_split_column_name` is provided but the fraction splits are not.

Adding `timestamp_split_column_name` to the conditional above should work.

Reply:
Okay, I thought they were required together, thanks. Will recheck the proto.

Reply:
Fixed